src/commands/mod.rs

```diff
@@ -30,6 +37,127 @@ pub enum Command {
     DeleteLinkdef(DeleteLinkdef),
 }
 
+impl Command {
+    /// Generates the corresponding event for a [Command](Command) in the form of a two-tuple ([Event](Event), Option<Event>)
+    ///
+    /// # Arguments
+    /// `model_name` - The model name that the command satisfies, needed to compute the proper annotations
+    ///
+    /// # Return
+    /// - The first element in the tuple corresponds to the "success" event a host would output after completing this command
+    /// - The second element in the tuple corresponds to an optional "failure" event that a host could output if processing fails
+    pub fn corresponding_event(&self, model_name: &str) -> Option<(Event, Option<Event>)> {
+        match self {
+            Command::StartActor(StartActor {
+                annotations,
+                reference,
+                host_id,
+                count,
+                ..
+            }) => {
+                let mut annotations = annotations.to_owned();
+                insert_managed_annotations(&mut annotations, model_name);
+                Some((
+                    Event::ActorsStarted(ActorsStarted {
+                        annotations: annotations.to_owned(),
+                        image_ref: reference.to_owned(),
+                        host_id: host_id.to_owned(),
+                        count: *count,
+                        // We do not know the public key or claims from the command
+                        public_key: String::with_capacity(0),
+                        claims: ActorClaims::default(),
+                    }),
+                    Some(Event::ActorsStartFailed(ActorsStartFailed {
+                        annotations,
+                        image_ref: reference.to_owned(),
+                        host_id: host_id.to_owned(),
+                        error: String::with_capacity(0),
+                        // We do not know the public key from the command
+                        public_key: String::with_capacity(0),
+                    })),
+                ))
+            }
+            Command::StopActor(StopActor {
+                annotations,
+                actor_id,
+                host_id,
+                count,
+                ..
+            }) => {
+                let mut annotations = annotations.to_owned();
+                insert_managed_annotations(&mut annotations, model_name);
+                Some((
+                    Event::ActorsStopped(ActorsStopped {
+                        annotations: annotations.to_owned(),
+                        public_key: actor_id.to_owned(),
+                        host_id: host_id.to_owned(),
+                        count: *count,
+                        // We don't know if there are additional instances, so can't consider remaining
+                        remaining: 0,
+                    }),
+                    None,
+                ))
+            }
+            Command::StartProvider(StartProvider {
```

Comment:
No `StopProvider` or `DeleteLinkdef`?

Reply:
I'm not sure what @brooksmtownsend was going for here, but my guess is that both of those don't matter for backing off. If the scaler was the one that stopped a provider, that is fine as it will get picked up by the scaler, see that it doesn't need to reconcile, and continue on. This is due to only one provider being able to run per link name + host combo.

The linkdef is similar, if we want to delete it, then it doesn't matter when we receive it as there can only ever be one linkdef (per link, provider, actor).